flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/observer/JobManagerDeploymentStatus.java

```diff
@@ -37,7 +37,11 @@
     DEPLOYING,
 
     /** JobManager deployment not found, probably not started or killed by user. */
-    MISSING;
+    // TODO: currently a mix of SUSPENDED and ERROR, needs cleanup
+    MISSING,
+
+    /** Deployment in terminal error, requires spec change for reconciliation to continue. */
+    ERROR;
```

Comment:
Should the `Observer#isReadyToReconcile` add the `ERROR` case branch?

Reply:
Good catch, done!